SU2_CFD/src/solvers/CNSSolver.cpp

```diff
@@ -613,6 +613,144 @@ su2double CNSSolver::GetCHTWallTemperature(const CConfig* config, unsigned short
+
+    SU2_OMP_FOR_DYN(OMP_MIN_SIZE)
+    for (auto iVertex = 0u; iVertex < geometry->nVertex[val_marker]; iVertex++) {
+
+      const auto iPoint = geometry->vertex[val_marker][iVertex]->GetNode();
+
+      if (!geometry->nodes->GetDomain(iPoint)) continue;
+
+      /*--- Compute dual-grid area and boundary normal ---*/
+
+      const auto Normal = geometry->vertex[val_marker][iVertex]->GetNormal();
+
+      su2double Area = GeometryToolbox::Norm(nDim, Normal);
+
+      su2double UnitNormal[MAXNDIM] = {0.0};
+      for (auto iDim = 0u; iDim < nDim; iDim++)
+        UnitNormal[iDim] = -Normal[iDim]/Area;
+
+      /*--- Compute closest normal neighbor ---*/
+
+      const auto Point_Normal = geometry->vertex[val_marker][iVertex]->GetNormal_Neighbor();
+
+      /*--- Get coordinates of i & nearest normal and compute distance ---*/
+
+      const auto Coord_i = geometry->nodes->GetCoord(iPoint);
+      const auto Coord_j = geometry->nodes->GetCoord(Point_Normal);
+
+      su2double dist_ij = GeometryToolbox::Distance(nDim, Coord_i, Coord_j);
+
+      /*--- Store the corrected velocity at the wall which will
+       be zero (v = 0), unless there is grid motion (v = u_wall)---*/
+
+      if (dynamic_grid) {
+        nodes->SetVelocity_Old(iPoint, geometry->nodes->GetGridVel(iPoint));
+      }
+      else {
+        su2double zero[MAXNDIM] = {0.0};
+        nodes->SetVelocity_Old(iPoint, zero);
+      }
+
+      for (auto iDim = 0u; iDim < nDim; iDim++)
+        LinSysRes(iPoint, iDim+1) = 0.0;
+      nodes->SetVel_ResTruncError_Zero(iPoint);
+
+      /*--- Get transport coefficients ---*/
+
+      su2double laminar_viscosity    = nodes->GetLaminarViscosity(iPoint);
+      su2double eddy_viscosity       = nodes->GetEddyViscosity(iPoint);
+      su2double thermal_conductivity = Cp * (laminar_viscosity/Prandtl_Lam + eddy_viscosity/Prandtl_Turb);
+
+      const su2double There = nodes->GetTemperature(Point_Normal);
+      su2double Twall = nodes->GetTemperature(iPoint);
+
+      /*--- Compute the normal gradient in temperature using Twall ---*/
+
+      su2double dTdn = -(There - Twall)/dist_ij;
+
+      /*--- Apply a weak boundary condition for the energy equation.
+       Compute the residual due to the prescribed heat flux. ---*/
+
+      su2double Res_Conv = 0.0;
+      su2double Res_Visc = Area * (thermal_conductivity * dTdn - emissivity*STEFAN_BOLTZMANN*pow(Twall,4));
+        
+      /*--- Calculate Jacobian for implicit time stepping ---*/
+
+      if (implicit) {
+
+        /*--- Add contributions to the Jacobian from the weak enforcement of the energy equations. ---*/
+
+        su2double Density = nodes->GetDensity(iPoint);
+        su2double Vel2 = GeometryToolbox::SquaredNorm(nDim, &nodes->GetPrimitive(iPoint)[prim_idx.Velocity()]);
+        su2double dTdrho = 1.0/Density * ( -Twall + (Gamma-1.0)/Gas_Constant*(Vel2/2.0) );
+
+        Jacobian_i[nDim+1][0] = thermal_conductivity/dist_ij * dTdrho * Area;
```

Comment:
You need the radiation part here. Or did it make things less stable? If so leave a comment please.